### ISSUE TITLE
[xml-model] formatting support

### ISSUE THREAD

**Issue by xorye** [eclipse/lemminx#713]
Currently, `<?xml-model href=""?>` formatting could be enhanced.

For example:
```
<?xml version="1.0" encoding="UTF-8"     ?>
<?xml-model href="http://www.docbook.org/xml/5.0/xsd/docbook.xsd"     ?>
```

Before formatting:
![image](https://user-images.githubusercontent.com/20326645/82598359-89e3a600-9b78-11ea-8965-b924fa27b3e2.png)

After formatting:
![image](https://user-images.githubusercontent.com/20326645/82598373-91a34a80-9b78-11ea-8339-0ad1a911b4ed.png)

In this case, `xml version` is being formatted correctly and `xml-model` could be better.

Two problems:
1. Formatting was done with `xml.preferences.quoteStyle = single` and `xml.enforce.quoteStyle = preferred`. The quotes do not change to single quotes for `xml-model`. 

2. There should be no space before `?>`.

**Comment by angelozerr** [comment]
I agree but the main problem is that href is not an attribute (the scanner doesn't consider href as attribute) because processing instruction can have any data content. perhaps we could improve the scanner to support attribute syntax but what about other processing instruction kind?
